weave-js/src/components/Form/TextField.tsx

```diff
@@ -78,34 +79,37 @@ export const TextField = ({
     <Tailwind style={{width: '100%'}}>
       <div
         className={classNames(
-          'relative',
-          textFieldSize === 'medium' ? 'h-32' : 'h-40'
+          'relative rounded-sm',
+          textFieldSize === 'medium' ? 'h-32' : 'h-40',
+          'outline outline-1 outline-moon-250',
+          {
+            'hover:outline-2 [&:hover:not(:focus-within)]:outline-[#83E4EB]':
+              !errorState,
+            'focus-within:outline-2 focus-within:outline-teal-400': !errorState,
+            'outline-2 outline-red-450': errorState,
+            'pointer-events-none opacity-50': disabled,
+          }
         )}>
-        <div
-          className={classNames(
-            'absolute bottom-0 top-0 flex w-full items-center rounded-sm',
-            'outline outline-1 outline-moon-250',
-            disabled
-              ? 'opacity-50'
-              : 'hover:outline hover:outline-2 hover:outline-teal-500/40 focus:outline-2',
-            errorState
-              ? 'outline outline-2 outline-red-450 hover:outline-red-450 focus:outline-red-450'
-              : 'outline outline-1 outline-moon-250 hover:outline-teal-500/40 focus:outline-teal-500/40'
```

Comment:
the reason outline wasn't working on focus is because we specified `:focus` styles on a `div`, but divs don't have focus state. the element being focused is actually the inner `input`. we must use `:focus-within` instead (see [above](https://github.com/wandb/weave/pull/2056/files#diff-cfe478a74185bdf97c39dd00bfa08d181cbf7bb6e9d24630ebad18b8f1b17bacR88))